mars/services/scheduling/supervisor/queueing.py

```diff
@@ -129,6 +131,10 @@ async def submit_subtasks(self, band: Tuple = None, limit: Optional[int] = None)
         if not limit and band not in self._band_slot_nums:
             self._band_slot_nums = await self._cluster_api.get_all_bands()
 
+        if band and band not in self._band_slot_nums:
+            await self.balance_queued_subtasks(from_band=band)
```

Comment:
Do we need to call `balance_queued_subtasks` here as it is already done in `watch_bands`?

Reply:
Maybe there are indeed some redundancies here.